nbqa/config.py

```diff
@@ -18,6 +23,31 @@ class _ConfigSections(NamedTuple):  # pylint: disable=R0903
 CONFIG_SECTIONS = _ConfigSections()
 
 
+try:
+    ISORT_MODULE: Optional[ModuleType] = import_module("isort")
+except ImportError:  # pragma: nocover (this case is tested in test_isort_works.py)
+    ISORT_MODULE = None
+
+
+def get_default_configs() -> Mapping[str, List[str]]:
+    """
+    Get default CLI args (if any) to call command with.
+
+    In the case of isort, we use ``--treat-comment-as-code '# %%'``.
+
+    Returns
+    -------
+    Mapping
+        Default configs for each code quality tool.
+    """
+    default_configs = defaultdict(lambda: [])
+    if ISORT_MODULE is not None:
```

Comment:
For the version check, we could use `importlib.metadata` with backporting lib for python 3.6 and 3.7.

```Python
from importlib.metadata import version

# maintain a dict of minimum versions supported for tools
min_ver_dict = {
"isort": "5.3.0"
}

# First thing we could after parsing the cli args is to check this in `main()`
command_version = version(command)
# this command version can be checked against in the min_ver_dict
# before even proceeding to run the entire procedure
```

## Reference

[importlib metadata usage](https://importlib-metadata.readthedocs.io/en/latest/using.html)

Reply:
I thinking using `import_metadata` for version checks as well as collecting any other metadata is much better than we doing it ourselves.

Also we can use the same version to decide if the package is installed or not and return immediately. This is better for performance since we are not actually importing any module.

Reply:
maybe this is better, actually...let's just "force" people to use isort 5.3.0+ and then we can always pass --treat-comment-as-code. Thanks for the suggestion, I'll make the changes!

Reply:
For passing the config, please consider using a default config file as suggested in the comment above.

Reply:
Sure

> Along with importlib.resources in Python 3.7 and newer (backported as importlib_resources for older versions of Python), this can eliminate the need to use the older and less efficient pkg_resources package.

I guess this means we need to use `pkg_resources` if we want to support Python3.6?

Reply:
There isn't any way of easily comparing versions in `importlib_metadata`, is there? Or do we still need `from pkg_resources import parse_version`?

Reply:
`importlib.metadata` is availalbe from python 3.8 onwards as part of stdlib. For python 3.6 and 3.7 we need to define `importlib_metadata; python_version < '3.8'` in setup.py `install_requires`.

## References
https://stackoverflow.com/questions/60517042/requirements-txt-conditional-on-python-version
https://pip.pypa.io/en/stable/reference/pip_install/#requirement-specifiers

Reply:
> There isn't any way of easily comparing versions in importlib_metadata, is there? Or do we still need from pkg_resources import parse_version?

Simple string comparison would work right? `"5.6.4" > "5.3.0"` would work.

Reply:
@girip11 I've just been trying this, but I don't think it works if we have a local module we want to import.

E.g. I get the following failures:
```
FAILED tests/test_runtime_errors.py::test_unable_to_reconstruct_message - ModuleNotFoundError: Command `remove_comments` not found by nbqa.
FAILED tests/test_runtime_errors.py::test_unable_to_reconstruct_message_pythonpath - ModuleNotFoundError: Command `remove_comments` not found by nbqa.
FAILED tests/tools/test_doctest.py::test_doctest_works - ModuleNotFoundError: Command `doctest` not found by nbqa.
FAILED tests/tools/test_doctest.py::test_doctest_invalid_import - ModuleNotFoundError: Command `doctest` not found by nbqa.
```

IMO we should allow for running local scripts / modules on notebooks, although I can't see how to tell if they're importable with trying
```
    try:
        import_module(command)
```
Any suggestions?

Reply:
In that case we have to distinguish between local script, standard module or python package. Shall we have option like `--script` to indicate it is a local script. This can allow for new modules that can be written by users too

Reply:
It should already work with custom modules though (e.g. see the test which runs `nbqa remove_comments`)

The overhead of importing modules doesn't seem very large
```
In [6]: %%timeit 
   ...: try: 
   ...:     import_module('foo') 
   ...: except: 
   ...:     pass 
   ...:                                                                                                                                                                                                   
58.8 µs ± 1.57 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```
is it worth adding to the API to reduce the overhead of using `import_module`?

Reply:
Adding these dependencies to nbqa at source code level which we claim to be tool_agnostic is not looking good when you have other ways right. Because then we would have more tool specific code creeping in to the source. So if we can go the generic way, even if its a hard way, its better to take it.

Reply:
Also I think its good to separate module from packages right? What are your thoughts?

Reply:
> Because then we would have more tool specific code creeping in to the source

If I've understood correctly, I think we'd need _some_ tool-specific code anyway, because of
```
>>> from importlib.metadata import version
>>> version('doctest')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.8/importlib/metadata.py", line 531, in version
    return distribution(distribution_name).version
  File "/usr/lib/python3.8/importlib/metadata.py", line 504, in distribution
    return Distribution.from_name(distribution_name)
  File "/usr/lib/python3.8/importlib/metadata.py", line 177, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: doctest
```

> Also I think its good to separate module from packages right?

As in, to have a different way of running `nbqa <installed package>` from `nbqa <local module>`? If so, I'm not sure that'd be desirable, because when running Python scripts you can do both `python -m <installed package>` and `python -m <local module>`

Reply:
> is it worth adding to the API to reduce the overhead of using import_module?

You are right. Hey how about using a fallback? 

```Python
# use importlib_metadata for checking version and module present or not.
# if we get result we can proceed straight way
# if we are not able to find version, try importlib.import_module
```

Reply:
Nice!! :+1: OK I'll do that, thanks!

Reply:
Thanks @MarcoGorelli . Sometimes I get carried away like saying `handle module and package differently`. I completely agree with you that its not desirable